### DISCUSSION TITLE
Why fields in same hierarchy get deleted when updating in Prisma?

### DISCUSSION THREAD

**Discussion by pawan-kr1997** [prisma/prisma#15820] [Q&A]
I am trying to implement updation using Prisma. here is my data structure.
```
{
  "name":"toy",
  "data":{
     "sports":{
        "currentState":"false",
        "oldState":"true"
    }
 }
  ```
Here is my logic to update currentState from false to true.
```
const updatedSource = await prisma.sources.update({
            where: {
                name: 'toy'
            },
            data: {
                data: {
                   update:{
                       sports: {
                           currentState: "true"
                       }
                   }
                }
            },
        })
```
Here is the schema file:
```
type SourcesData {
  sports        SourcesDataState
}

type SourcesDataState {
  currentState String
  oldState     String
}


model sources {
  id           String   @id @default(auto()) @map("_id") @db.ObjectId
  data         SourcesData
  name         String   @unique
}
```

The above logic updates the currentState field but deletes the oldState field from the database.

Please guide me on how to update ``currentState`` in a way that ``oldState`` data persists.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
